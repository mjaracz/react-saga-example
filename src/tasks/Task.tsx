import React, { FC, Fragment } from 'react'
import { ITaskProps } from './interfaces/task.interface'
import { Collapse, List, ListItem, ListItemIcon, ListItemText } from '@material-ui/core'
import { ExpandLess, ExpandMore, AccessAlarm, Delete } from '@material-ui/icons'
import { useTask } from './hooks/useTask'
import './styles/task.css'

export const Task: FC<ITaskProps> = ({ title, description, id }) => {
  const { isOpenDesc, handleOpenDesc, handleDeleteTask } = useTask()
  return (
    <Fragment>
      <ListItem className="tasks__item" >
        <ListItemIcon>
          <AccessAlarm />
        </ListItemIcon>
        <ListItemText className="item__text" primary={title} />
        {
          description &&
          <div className="item__expend" onClick={handleOpenDesc}>
		        <span className="expend__text">description</span>
            { isOpenDesc ? <ExpandLess/> : <ExpandMore/> }
	        </div>
        }
        <ListItemIcon className="item__deleteIcon" onClick={() => handleDeleteTask(id)} >
          <Delete/>
        </ListItemIcon>
      </ListItem>
      <Collapse in={isOpenDesc} timeout="auto" unmountOnExit>
        <List className="tasks__item--expended" component="div" disablePadding>
          <ListItem>
            <ListItemText primary={description} />
          </ListItem>
        </List>
      </Collapse>
    </Fragment>
  )
}
