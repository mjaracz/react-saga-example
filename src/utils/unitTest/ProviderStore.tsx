import React, { FC } from 'react'
import { Provider } from 'react-redux'
import { store } from '../../redux/store'

export const ProviderStore: FC = ({ children }) => {
  return (
    <Provider store={store}>
      { children }
    </Provider>
  )
}
