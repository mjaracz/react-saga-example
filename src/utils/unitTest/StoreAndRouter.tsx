import React, { FC } from 'react'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'
import { store } from '../../redux/store'

export const StoreAndRouter: FC = ({ children }) => {
  return (
    <Provider store={store}>
      <BrowserRouter>
        { children }
      </BrowserRouter>
    </Provider>
  )
}
