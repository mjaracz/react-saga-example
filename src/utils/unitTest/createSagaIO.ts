import { stdChannel } from 'redux-saga'
import { EventEmitter } from 'events'


export const createSagaIO = (emitter: EventEmitter) => {
  const channel = stdChannel()
  emitter.on('action', channel.put)

  return {
    channel,
    dispatch: action => emitter.emit('action', action),
    getState: () => ({})
  }
}
