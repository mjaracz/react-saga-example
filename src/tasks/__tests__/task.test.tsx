import React from 'react'
import { render } from '@testing-library/react'
import { Task } from '../Task'
import { StoreAndRouter } from '../../utils/unitTest/StoreAndRouter'

describe('<Task />', () => {
  it('should match the snapshot', () => {
    const { baseElement } = render(
      <StoreAndRouter>
        <Task title='title example' id={0} />
      </StoreAndRouter>
    )
    expect(baseElement).toMatchSnapshot()
  })
})
