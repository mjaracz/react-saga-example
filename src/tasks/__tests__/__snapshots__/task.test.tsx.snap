// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Task /> should match the snapshot 1`] = `
<body>
  <div>
    <li
      class="MuiListItem-root tasks__item MuiListItem-gutters"
    >
      <div
        class="MuiListItemIcon-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M22 5.72l-4.6-3.86-1.29 1.53 4.6 3.86L22 5.72zM7.88 3.39L6.6 1.86 2 5.71l1.29 1.53 4.59-3.85zM12.5 8H11v6l4.75 2.85.75-1.23-4-2.37V8zM12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9c4.97 0 9-4.03 9-9s-4.03-9-9-9zm0 16c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7z"
          />
        </svg>
      </div>
      <div
        class="MuiListItemText-root item__text"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          title example
        </span>
      </div>
      <div
        class="MuiListItemIcon-root item__deleteIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </div>
    </li>
  </div>
</body>
`;
